core/moddeps.go

```diff
@@ -46,10 +46,14 @@ type ModDeps struct {
 func NewModDeps(root *Query, mods []Mod) *ModDeps {
 	return &ModDeps{
 		root: root,
-		Mods: mods,
+		Mods: append([]Mod{}, mods...),
```

Comment:
Lol, this was fun :cry: What was happening was that the `defaultDeps` were being modified - then this affected all the other module loading. To avoid this, we just need to be *very* careful, and always copy the input.